txpipe/covariance.py

```diff
@@ -788,12 +813,22 @@ def get_workspaces_dict(self, cl_sacc, masks_names):
                 hashes[m] = cl_sacc.metadata[f"hash/{m}"]
         ell_hash = cl_sacc.metadata["hash/ell_hash"]
 
-        w = {}
+        w = {'00': {}, '02': {}, '22': {}}
+        # Get the number of data points per Cell
+        dtype = cl_sacc.get_data_types()[0]
+        trs = cl_sacc.get_tracer_combinations()[0]
+        ell_eff, _ = cl_sacc.get_ell_cl(dtype, *trs)
+        n_ell = ell_eff.size
         for tr1, tr2 in cl_sacc.get_tracer_combinations():
+            # This assumes that the name of the tracers will be 'lens'or
+            # 'source'
+            s1 = 0 if 'lens' in tr1 else 2
+            s2 = 0 if 'lens' in tr2 else 2
+            sk = ''.join(sorted(f'{s1}{s2}'))
             m1 = masks_names[tr1]
             m2 = masks_names[tr2]
             key = (m1, m2)
-            if (key in w) or (key[::-1] in w):
+            if (key in w[sk]) or (key[::-1] in w[sk]):
```

Comment:
Not sure what I was doing here with the `key[::-1]` part.  If you know could you add a comment!

Reply:
I think key[::-1] reverses the order so you were probably checking m1, m2 or m2, m1 combinations right? I added a comment.

Reply:
Thanks!